erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -812,9 +812,22 @@ def validate_serial_numbers(self):
 		"""
 			validate serial number agains Delivery Note and Sales Invoice
 		"""
+		self.set_serial_no_against_delivery_note()
 		self.validate_serial_against_delivery_note()
 		self.validate_serial_against_sales_invoice()
 
+	def set_serial_no_against_delivery_note(self):
```

Comment:
The serial nos should not be set forcefully on validate. The user should be able to change the serial nos. It should be set via mapper while creating Invoice from DN.

And serial nos will validated against DN, to ensure the serial no exists in DN.